src/Symfony/Component/Translation/Translator.php

```diff
@@ -213,7 +213,13 @@ public function trans($id, array $parameters = [], $domain = null, $locale = nul
         }
 
         if ($this->hasIntlFormatter && $catalogue->defines($id, $domain.MessageCatalogue::INTL_DOMAIN_SUFFIX)) {
-            return $this->formatter->formatIntl($catalogue->get($id, $domain), $locale, $parameters);
+            $message = $catalogue->get($id, $domain);
+            // formatIntl uses MessageFormatter which constructor throws an exception if the message is empty
+            if ('' === $message) {
+                return '';
+            }
```

Comment:
I think this check should be moved to `IntlFormatter::formatIntl()` method, otherwise anyone using it will face the same problem.